python_ta/checkers/unnecessary_indexing_checker.py

```diff
@@ -87,16 +88,38 @@ def _iterable_if_range(node: NodeNG) -> Optional[str]:
 def _is_load_subscript(index_node: astroid.Name, for_node: astroid.For) -> bool:
     """Return whether or not <index_node> is used to subscript the iterable of <for_node>
     and the subscript item is being loaded from, e.g., s += iterable[index_node].
+
+    NOTE: Index node is deprecated in Python 3.9
+
+    Returns True if the following conditions are met:
+    (3.9)
+        - The <index_node> Name node is inside of a Subscript node
+        - The item that is being indexed is the iterable of the for loop
+        - The Subscript node is being used in a load context
+    (3.8)
+        - The <index_node> Name node is inside of an Index node
+        - The Index node is inside of a Subscript node
+        - The item that is being indexed is the iterable of the for loop
+        - The Subscript node is being used in a load context
     """
     iterable = _iterable_if_range(for_node.iter)
-    # Name node is not inside Subscript node iterable[index].
-    if not (isinstance(index_node.parent, astroid.Index) and isinstance(index_node.parent.parent, astroid.Subscript)
-            and isinstance(index_node.parent.parent.value, astroid.Name)
-            and index_node.parent.parent.value.name == iterable):
-        return False
-    # Use ctx attribute to find out what context the subscript is being used in.
-    subscript_node = index_node.parent.parent
-    return subscript_node.ctx == astroid.Load
+    version = sys.version[:3]
```

Comment:
You're on the right track, but this is a less common way of checking versions, especially since we actually want to check for versions >= 3.9.

Use [`sys.version_info`](https://docs.python.org/3/library/sys.html#sys.version_info) and tuple comparison in the check below.